datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1478,6 +1480,53 @@ macro_rules! to_string {
     }};
 }
 
+
+/// Array_union SQL function
+pub fn array_union(args: &[ArrayRef]) -> Result<ArrayRef> {
+    if args.len() != 2 {
+        return exec_err!("array_union needs two arguments")
+    }
+    let array1 = &args[0];
+    let array2= &args[1];
+
+    check_datatypes("array_union", &[array1, array2])?;
+    let list1 = as_list_array(array1)?;
+    let list2 = as_list_array(array2)?;
+    eprintln!("{:?}",list1.value_type());
+    match (list1.value_type(), list2.value_type()){
+        (DataType::Null, _) => {
+            Ok(array2.clone())
+        },
+        (_, DataType::Null) => {
+            Ok(array1.clone())
+        }
+        (DataType::List(_), DataType::List(_)) => {
+            let result = concat_internal(args)?;
+            let row_converter = RowConverter::new(vec![
+                SortField::new(
+                    args[0].data_type().clone()
+                )
+            ]
+            )?;
+            let converted = row_converter.convert_columns(&[result])?;
```

Comment:
I think we can simply get the internal value via iterating the array

```rust
        let list_arr = as_list_array(&arr).unwrap();
        let row_number = list_arr.len();
        for i in 0..row_number {
            let arr = list_arr.value(i);
            let i64arr = as_primitive_array::<Int64Type>(&arr);
            for v in i64arr.iter() {
                // v is Option<i64>
            }
        }
```

I'm not so familiar with RowConverter and Row, maybe it is more efficient. But I think we can try out the correct implementation first and then improve on it latter

Reply:
```rust
let arr1 = ListArray::from_iter_primitive::<Int64Type, _, _>(vec![
            Some(vec![Some(1), Some(2), Some(3)]),
            Some(vec![Some(4), Some(5)]),
            Some(vec![Some(7)]),
        ]);

let arr2 = ListArray::from_iter_primitive::<Int64Type, _, _>(vec![
            Some(vec![Some(4)]),
            Some(vec![Some(4), Some(5)]),
            Some(vec![Some(7), Some(8), Some(9)]),
        ]);

let arr = concat_internal(&[Arc::new(arr1), Arc::new(arr2)]).unwrap();
```

```txt
arr: ListArray
[
  PrimitiveArray<Int64> // 1st row
[
  1,
  2,
  3,
  4,
],
  PrimitiveArray<Int64> // 2nd row
[
  4,
  5,
  4,
  5,
],
  PrimitiveArray<Int64> // 3rd row
[
  7,
  7,
  8,
  9,
],
]
```
Expected array after union.
```txt 
arr: ListArray
[
  PrimitiveArray<Int64> // 1st row
[
  1,
  2,
  3,
  4,
],
  PrimitiveArray<Int64> // 2nd row
[
  4,
  5,
],
  PrimitiveArray<Int64> // 3rd row
[
  7,
  8,
  9,
],
]
```

Reply:
> ```rust
> arrow::compute::concat(arrays.as_slice())?;
> ```

You may need concat_internal